core/charm/repository/charmhub.go

```diff
@@ -953,14 +953,23 @@ func refreshConfig(charmURL *charm.URL, origin corecharm.Origin) (charmhub.Refre
 		method = MethodID
 	}
 
+	var track string
+	if origin.Platform.Channel != "" {
+		var err error
+		track, err = corecharm.ChannelTrack(origin.Platform.Channel)
```

Comment:
Use a base channel type, not a charm channel type. They are not equivalent. 
corebase.ParseChannel